xesmf/frontend.py

```diff
@@ -895,6 +953,85 @@ def __init__(
         else:
             self.out_coords = {lat_out.name: lat_out, lon_out.name: lon_out}
 
+        if parallel:
+            # Ensure ds_in is not dask-backed
+            if xr.core.pycompat.is_dask_collection(ds_in):
+                ds_in = ds_in.compute()
```

Comment:
This seems dangerous no? I guess you mean to load the coordinates only ? This call would load all data variables too, even though we don't need them.

I'd put this line further down the line, when the extra stuff is dropped.